service/src/main/java/org/cbioportal/service/impl/ClinicalAttributeServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package org.cbioportal.service.impl;
+
+import org.cbioportal.model.ClinicalAttribute;
+import org.cbioportal.model.meta.BaseMeta;
+import org.cbioportal.persistence.ClinicalAttributeRepository;
+import org.cbioportal.service.ClinicalAttributeService;
+import org.cbioportal.service.exception.ClinicalAttributeNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PostFilter;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ClinicalAttributeServiceImpl implements ClinicalAttributeService {
+
+    @Autowired
+    private ClinicalAttributeRepository clinicalAttributeRepository;
+
+    @Override
+    @PostFilter("hasPermission(filterObject.cancerStudyIdentifier, 'CancerStudy', 'read')")
```

Comment:
Here, it would be nicer to be able to pass a lambda function or something similar instead of string.